### PR TITLE
[Multi-asic]: Fix function to get number of npus to parse updated asic.conf file

### PR DESCRIPTION
**Pull request by SuvarnaMeenakshi**
Signed-off-by: SuvarnaMeenakshi <sumeenak@microsoft.com>

### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [X] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
Function to get number of asics on a platform does not return the correct number of asics because the asic.conf file format was modified.
Function to generate the multi-instance service names running on multi-asic platform fails.
#### How did you do it?
- Fix the function to get the number of asics by parsing asic.conf, update the function to parse the file and get number of asics against "NUM_ASIC" keyword.
- Fix the function to generate the multi-instance service names.
- Remove usage of num_npu in bgp_facts test case as it is not used currently.

#### How did you verify/test it?
Ran any multi-asic compatible test case(SNMP) on multi-asic platform, test runs successfully.
Single-asic platform - No change.

### CODE REVIEW

File: tests/common/devices.py [hunk 1]
```diff
@@ -213,19 +213,23 @@ def _get_npu_count(self, platform):
         """
         Gets the number of npus for this device.
         """
-
+        num_npu = 1
```
Comment (abdosi) [inline tests/common/devices.py:216]: @SuvarnaMeenakshi  Good opportunity to make make npu asic.
We have to fix that in all places going forward.

Comment (SuvarnaMeenakshi) [reply tests/common/devices.py]: Updated as suggested.

File: tests/bgp/test_bgp_fact.py [hunk 2]
```diff
@@ -8,7 +8,6 @@
 def test_bgp_facts(duthost):
     """compare the bgp facts between observed states and target state"""
 
-    npus = duthost.num_npus()
```
Comment (abdosi) [inline tests/bgp/test_bgp_fact.py:11]: not needed anymore ?

Comment (SuvarnaMeenakshi) [reply tests/bgp/test_bgp_fact.py:11]: npus variable is not used in this file, it was added for multi-asic support, but was removed by another PR.
So, will remove it for now and will be added back in future for supporting multi-asic testing.

**Pull request merged**